pennylane/ops/functions/eigvals.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the qml.eigvals function.
+"""
+# pylint: disable=protected-access
+from functools import reduce
+import warnings
+
+import pennylane as qml
+
+
+@qml.op_transform
+def eigvals(op):
+    r"""The matrix representation of an operation or quantum circuit.

```

Comment:
```suggestion
    r"""The eigenvalues of an operation or quantum circuit.
```

Reply:
I'm noting this only now, but the eigenvals/matrix of a circuit is not well defined, no? Wouldn't it always be a sequence of ops (qfunc, tape or whatever it comes as)?

Reply:
```suggestion
    r"""The eigenvalues of one or more operations.
```